process/smartContract/process.go

```diff
@@ -571,6 +586,7 @@ func (sc *scProcessor) ExecuteBuiltInFunction(
 		log.Debug("processed built in functions error", "error", err.Error())
 		return 0, err
 	}
+	builtInFuncGasUsed := vmInput.GasProvided - vmOutput.GasRemaining
```

Comment:
Above, we use `safeSubUint64`.

```
consumedGas := safeSubUint64(tx.GetGasLimit(), vmOutput.GasRemaining)
```

though here we do not. Not needed (because the VM will not output incorrect values, such as a gas remaining higher than gas provided)?

Reply:
this should be impossible